### ISSUE TITLE
Feature Request: Data source to parse an ARN

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#3995]
Add a data source to parse an ARN into its [constituent parts](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#genref-arns).

**Comment by bflad** [comment]
Can you explain the use case?

**Comment by barend** [comment]
It's a convenient way to get to the account ID and region of any resource. I find myself using this from time to time, for example when constructing an IAM policy document:

```hcl
foo_arn = "arn:aws:xxx::${element(split(":", data.aws_yyy.zzz.arn), 4)}:foo/bar"
```

It would help readability if I could use something akin to

```hcl
data "aws_arn" "xxx" {
  arn = "${data.aws_yyy.zzz.arn}"
}

resource "aws_abc" "abc" {
  foo_arn = "arn:aws:xxx::${data.aws_arn.xxx.account_id}:foo/bar"
}
```

**Comment by ewbankkit** [comment]
Yes, my specific use case is some prototype work around the Resource Groups Tagging [`GetResources`](https://docs.aws.amazon.com/resourcegroupstagging/latest/APIReference/API_GetResources.html) API which returns a list of ARNs. I'd like to be able to parse the ARN to extract resource type etc.

**Comment by bflad** [comment]
This will be released later this week in v1.19.0 of the AWS provider.

**Comment by bflad** [comment]
This has been released in [version 1.19.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1190-may-16-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
